pkg/apis/middleware/session.go

```diff
@@ -24,6 +24,9 @@ func CreateTokenToSessionFunc(verify VerifyFunc) TokenToSessionFunc {
 			Email             string `json:"email"`
 			Verified          *bool  `json:"email_verified"`
 			PreferredUsername string `json:"preferred_username"`
+                        // TODO:  It's unclear whether the claim from --oidc-groups-claim should be here.
+                        //        It appears to be hard-coded elsewhere in oauth2-proxy.
+                        Groups            []string `json:"groups,omitempty"`
```

Comment:
This isn't globally compatible (since `groups` technically isn't an official claim in the OIDC spec). If you peek at the history of the OIDCProvider PRs, you'll see we had to iterate on this many times to be compatible with various OIDC IdPs that didn't follow the spec.

https://github.com/oauth2-proxy/oauth2-proxy/blob/c00d36159e9b55c42554593e6bf305e706acb5b3/providers/provider_data.go#L211

But that is a private method for providers, it isn't exposed to this generic area.

Reply:
Would it resolve your concern if I used the claim from --oidc-groups-claim here?

Certainly, nobody is currently using `--extra-jwt-providers` and `--allowed-group` together, because if they were then no one would ever gain access through any of the extra JWT providers:  the groups from these providers would never be recognized and therefore all authorization attempts would fail.  That, at least, implies that one of two things should happen:

1)  Use of the `--extra-jwt-providers` and `--allowed-group` flags should be mutually exclusive, or
2)  The groups from the `--extra-jwt-providers` should be recognized.

Reply:
I added more context in the issue -- let's discuss more there. This is a very buggy/risky area of the code for compatibility & we have another issue laying out how we want to replace this architecture in a more scalable way moving forward.

Reply:
@theGOTOguy @NickMeves We have similar use case and problem than can be solved by this PR - we need authentication and authorization based on groups for machines and humans. I consider this project as great to support both types of communication. Moreover, our identity provider supporting groups in JWT access tokens so they are not needed to request just copied from the token.

 It is pity it was not merged as something that can be used before reworking of authorization.